CAN-D/Src/stm32f3xx_it.c

```diff
@@ -114,11 +121,45 @@ void TIM2_IRQHandler(void)
 }
 
 /**
-  * @brief This function handles USART2 global interrupt / USART2 wake-up interrupt through EXTI line 26.
+  * @brief This function handles USART2 global interrupt,
+  *        USART2 RX interrupts, and USART2 wake-up interrupt 
+  *        through EXTI line 26.
   */
 void USART2_IRQHandler(void)
 {
+  char rxData[128] = "0";
+  uint8_t rx_idx = 0;
+
   HAL_UART_IRQHandler(&huart);
+
+  // Check if the UART2 Read Data Register has data
+  if (huart.Instance->ISR & USART_ISR_RXNE)
+  {
+    // Read the data from the register
+    rxData[rx_idx] = huart.Instance->RDR;
+
+    // The GPS RX data will be held between '$' and '\n' characters
+    if (rxData[rx_idx] == '$')
+    {
+      while (rxData[rx_idx] != '\n')

```

Comment:
I'm not sure I quite understand what is happening here or what the purpose of rx_idx is.

Reply:
Oh right haha I was planning on tracking the index as a means of tracking the length, in bytes, of the transmission data. But I don't think it's necessary because the GPS messages should all be the same length. So I'll remove this logic.